### PR TITLE
Refactor to enumerated types

### PR DESCRIPTION
**Pull request by simonsso**
Boolean options replaced with an enumerated types
to improve readability of the code.

### CODE REVIEW

File: core/src/state.rs [hunk 1]
```diff
@@ -97,6 +97,18 @@ pub enum State {
     Live(LiveState),
 }
 
+#[derive(PartialEq, PartialOrd)]
+pub enum StateRuntimeCheck {
+    TryRuntimeCheck,
+    NoTryRuntimeCheck,
+}
```
Comment (liamaharon) [inline core/src/state.rs:104]:
```suggestion
#[derive(PartialEq, PartialOrd)]
pub enum TryRuntimeFeatureCheck {
    Check,
    Skip,
}
```

File: core/src/state.rs [hunk 2]
```diff
@@ -97,6 +97,18 @@ pub enum State {
...
+
+#[derive(PartialEq, PartialOrd)]
+pub enum StateSpecVersionCheck {
+    CheckSpecVersion,
+    NoSpecVersionCheck,
+}
```
Comment (liamaharon) [inline core/src/state.rs:110]:
```suggestion
#[derive(PartialEq, PartialOrd)]
pub enum SpecVersionCheck {
    Check,
    Skip,
}
```

File: rustfmt.toml [hunk 3]
```diff
@@ -1,13 +1,7 @@
...
-imports_granularity = "Crate"
-group_imports = "StdExternalCrate"
 reorder_imports = true
 
 # Consistency
 newline_style = "Unix"
-
-# Format comments
-comment_width = 100
-wrap_comments = true
```
Comment (liamaharon) [inline rustfmt.toml:13]: We enforce nightly fmt checks in polkadot-sdk and I'd like to keep that here too. 

I think instead we need to modify the `rust-toolchain.toml` to be something like
```toml
[toolchain]
channel = "nightly"
components = ["rustfmt"]
```

**Review by liamaharon — COMMENTED** [review]
Thanks for this! Just some minor comments.

File: rustfmt.toml (hunk 3 above)
Comment (simonsso) [reply rustfmt.toml:13]: Aha I will remove this from this PR as it is not related.

**Comment by simonsso** [comment]
> Thanks for this! Just some minor comments.

Thanks for your very useful comments, I have updated PR,
Sorry for the force push but I was not sure if you would merge or squash the commit for a nice history.

File: core/src/state.rs [hunk 4]
```diff
@@ -97,6 +97,25 @@ pub enum State {
...
 }
 
+/// Options for [`to_ext`]
+///
+/// Check if runtime is compiled with try-runtime option or not
+#[derive(PartialEq, PartialOrd)]
+pub enum TryRuntimeFeatureCheck {
+    Check,
+    Skip,
+}
```
Comment (liamaharon) [inline core/src/state.rs:107]:
```suggestion
/// Whether to check that the runtime was compiled with try-runtime feature
#[derive(PartialEq, PartialOrd)]
pub enum TryRuntimeFeatureCheck {
    /// Check the runtime was compiled with try-runtime feature
    Check,
    /// Don't check if the runtime was compiled with try-runtime feature
    Skip,
}
```

File: core/src/state.rs [hunk 5]
```diff
@@ -97,6 +97,25 @@ pub enum State {
...
+/// Options for [`to_ext`]
+///
+/// Enable of disable check wether new runtime spec version is not greater
+/// than the on-chain runtime spec version in [`to_ext`]
+
+#[derive(PartialEq, PartialOrd)]
+pub enum SpecVersionCheck {
+    Check,
+    Skip,
+}
```
Comment (liamaharon) [inline core/src/state.rs:117]: Needs to be wrapped at col 100
```suggestion
/// Whether to check if the new runtime `spec_version` is greater than the previous runtime `spec_version`
#[derive(PartialEq, PartialOrd)]
pub enum SpecVersionCheck {
    /// Check that the new runtime `spec_version` is greater than the previous runtime `spec_version`
    Check,
    /// Don't check that the new runtime `spec_version` is greater than the previous runtime `spec_version`
    Skip,
}
```

**Review by liamaharon — APPROVED** [review]

**Comment by liamaharon** [comment]
> Sorry for the force push but I was not sure if you would merge or squash the commit for a nice history.

No worries. I'll squash merge, so feel free to push as many commits as you like here.

**Comment by liamaharon** [comment]
@simonsso could you please merge master into here so we can make sure it passes your new CICD checks?

**Pull request merged**